drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,481 @@
+
+	/**
+	 * Configuration:
+	 * - bits [0-2] : panel driver settling time
+	 * - bits [3-5] : touch detect delay
+	 * - bits [6-7] : touch average control
+	 */
+
+	return config->panel_driver_settling_time_us |
+	       config->touch_detect_delay_us    << 3 |
+	       config->touch_average_control << 6;
+}
+
+static int stmpe811_ts_init(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	int err;
+	uint8_t tmp;
+
+	err = stmpe811_reset(dev);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Get the current register value */
+	err += i2c_reg_read_byte_dt(&config->bus, STMPE811_SYS_CTRL2_REG, &tmp);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Select TSC pins in TSC alternate mode */
+	err += stmpe811_io_enable_af(dev, STMPE811_TOUCH_IO_ALL);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the functionalities to be enabled */
+	tmp &= ~(STMPE811_IO_FCT | STMPE811_TS_FCT | STMPE811_ADC_FCT);
+
+	/* Set the new register value */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_SYS_CTRL2_REG, tmp);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Select sample time, bit number and ADC reference */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_ADC_CTRL1_REG, ADC_CONF);
+	if (err < 0) {
+		return err;
+	}
+
+	k_msleep(STMPE811_WAIT_DELAY_MS);
+
+	/* Select the ADC clock speed */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_ADC_CTRL2_REG, ADC_CLOCK_SPEED);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Touch Screen Control config */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_CFG_REG,
+				     stmpe811_tsc_config_bits(dev));
+	if (err < 0) {
+		return err;
+	}
+
+	/* Configure the touch FIFO threshold */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_TH_REG, FIFO_TH_SINGLE_POINT);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Clear the FIFO memory content */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG, FIFO_STA_CLEAR);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Put the FIFO back into operation mode */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG, FIFO_STA_OPERATIONAL);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the range and accuracy of the pressure measurement (Z) */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_FRACT_XYZ_REG,
+				     PRESSURE_MEASUREMENT_CONF);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Set the driving capability (limit) of the device for TSC pins */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_I_DRIVE_REG, TSC_I_DRIVE_LIMIT);
+	if (err < 0) {
+		return err;
+	}
+
+	/* Touch screen control configuration */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_TSC_CTRL_REG, TSC_CONF);
+	if (err < 0) {
+		return err;
+	}
+
+	/**
+	 * Clear all the status pending bits if any.
+	 * Writing '1' to this register clears the corresponding bits.
+	 * This is an 8-bit register, so writing 0xFF clears all.
+	 */
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_INT_STA_REG, BIT_MASK(8));
```

Comment:
add the error exit path here as well and then the one on 197 becomes a `return 0`, just for consistency and one less thing to keep track of if you end up adding transactions to this code or if it ends up being refactored around